api/queues.test.js

```diff
@@ -220,6 +220,156 @@ describe('Queues API', () => {
     })
   })
 
+  describe('PATCH /api/queues/:queueId', () => {
+    test('succeeds for course staff with well-formed request no location', async () => {
+      const attributes = { name: 'CS 225 Queue 1 Alter', location: '' }
+      const res = await request(app)
+        .patch('/api/queues/1?forceuser=225staff')
+        .send(attributes)
+      expect(res.statusCode).toBe(201)
+      expect(res.body.name).toBe('CS 225 Queue 1 Alter')
+      expect(res.body.id).toBe(1)
+      expect(res.body.location).toBe('')
+    })
+
+    test('succeeds for course staff with well-formed request with location', async () => {
+      const attributes = { name: 'CS 225 Queue 1 Alter', location: 'Where' }
+      const res = await request(app)
+        .patch('/api/queues/1?forceuser=225staff')
+        .send(attributes)
+      expect(res.statusCode).toBe(201)
+      expect(res.body.name).toBe('CS 225 Queue 1 Alter')
+      expect(res.body.id).toBe(1)
+      expect(res.body.location).toBe('Where')
+    })
+
+    test('succeeds for admin with well-formed request no location', async () => {
+      const attributes = { name: 'CS 225 Queue 1 Alter', location: '' }
+      const res = await request(app)
+        .patch('/api/queues/1?forceuser=admin')
+        .send(attributes)
+      expect(res.statusCode).toBe(201)
+      expect(res.body.name).toBe('CS 225 Queue 1 Alter')
+      expect(res.body.id).toBe(1)
+      expect(res.body.location).toBe('')
+    })
+
+    test('succeeds for admin with well-formed request with location', async () => {
+      const attributes = { name: 'CS 225 Queue 1 Alter', location: 'Where' }
+      const res = await request(app)
+        .patch('/api/queues/1?forceuser=admin')
+        .send(attributes)
+      expect(res.statusCode).toBe(201)
+      expect(res.body.name).toBe('CS 225 Queue 1 Alter')
+      expect(res.body.id).toBe(1)
+      expect(res.body.location).toBe('Where')
+    })
+
+    test('fails for course staff with ill-formed request no location', async () => {
```

Comment:
It's probably safe just to test that it fails with a 403 with a well-formed request; if that's the case, it shouldn't ever succeed for an ill-formed request either.

Reply:
Sure, do you think it will be necessary to also test for fixed location queue here?

Reply:
Yeah, probably wouldn't hurt to make sure you can still update the location for a fixed-location one.

Reply:
Done. Please let me know if I'm being too paranoid when writing test cases.